manifests/charts/istiod-remote/templates/clusterrole.yaml

```diff
@@ -36,6 +36,19 @@ rules:
     verbs: [ "get", "watch", "list", "update", "patch", "create", "delete" ]
     resources: [ "workloadentries/status" ]
 
+  # gateway controller
+{{- if not (eq (toString .Values.pilot.env.PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER) "false") }}
```

Comment:
If possible, can you add a new ClusterRole object with the extra stuff, and a separate ClusterRoleBinding ? 
Seems better than having cluster roles that have different content based on flags.

Usually the cluster roles should be narrow and documented - the SA can have multiple bindings.